src/components/Tasks/Toolbar.tsx

```diff
@@ -23,16 +29,21 @@ interface PropsType {
 export default function Toolbar(props: PropsType) {
   const { defaultCollection, onItemSave, showCompleted, setShowCompleted } = props;
 
-  const [anchorEl, setAnchorEl] = React.useState<null | HTMLElement>(null);
+  const [sortAnchorEl, setSortAnchorEl] = React.useState<null | HTMLElement>(null);
+  const [optionsAnchorEl, setOptionsAnchorEl] = React.useState<null | HTMLElement>(null);
 
-  const handleClick = (event: React.MouseEvent<HTMLElement>) => {
-    setAnchorEl(event.currentTarget);
-  };
+  const dispatch = useDispatch();
+  const taskSettings = useSelector((state: StoreState) => state.settings.taskSettings);
+  const { sortBy } = taskSettings;
 
-  const handleClose = () => {
-    setAnchorEl(null);
+  const handleSortChange = (sort: string) => {
+    dispatch(setSettings({ taskSettings: { ...taskSettings, sortBy: sort } }));
```

Comment:
Please also close the menu here (only for the sorting, not for the options menu).